lib/Sema/TypeCheckDecl.cpp

```diff
@@ -4798,6 +4798,30 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
     return hadError;
   }
 
+  static TypeLoc getTypeLocForFunctionResult(FuncDecl *FD) {
```

Comment:
Is this necessary? I think it's OK if the attribute does not have a source location, and is marked implicit instead

Reply:
I'm not sure I follow. I'm using this to get at the appropriate `TypeRepr` to see if it's a `TypeReprKind::ImplicitlyUnwrappedOptional`.

Reply:
Sorry, I misread the the code.